stdlib/LibGit2/src/gitcredential.jl

```diff
@@ -7,11 +7,11 @@ Git credential information used in communication with git credential helpers. Th
 named using the [input/output key specification](https://git-scm.com/docs/git-credential#IOFMT).
 """
 mutable struct GitCredential
-    protocol::Union{String, Nothing}
-    host::Union{String, Nothing}
-    path::Union{String, Nothing}
-    username::Union{String, Nothing}
-    password::Union{String, Nothing}
+    protocol::Union{AbstractString, Nothing}
+    host::Union{AbstractString, Nothing}
+    path::Union{AbstractString, Nothing}
+    username::Union{AbstractString, Nothing}
+    password::Union{Base.SecretBuffer, Nothing}
```

Comment:
These shouldn't be AbstractString – we may need to convert them eventually anyways, and we may be using this as the GC root for them.